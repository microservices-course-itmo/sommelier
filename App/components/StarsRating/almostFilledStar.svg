<svg width="17" height="15" viewBox="0 0 17 15" fill="none" xmlns="http://www.w3.org/2000/svg">
    <linearGradient id="grad">
        <stop offset="50%" stop-color="#ECAB2E"/>
        <stop offset="50%" stop-color="grey" stop-opacity="0"/>
    </linearGradient>
    <path d="M8.5 0L10.2822 5.48499H16.0494L11.3836 8.8749L13.1658 14.3599L8.5 10.97L3.83419 14.3599L5.61637 8.8749L0.950553 5.48499H6.71782L8.5 0Z" fill="url(#grad)" stroke="#ECAB2E"/>
</svg>
