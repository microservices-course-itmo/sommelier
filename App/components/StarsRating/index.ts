export { default as StarsRating } from './starsRating'
