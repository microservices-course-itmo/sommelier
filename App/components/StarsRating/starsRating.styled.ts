import styled from 'styled-components/native'

export const container = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`

export const ratingContainer = styled.View`
  flex-direction: row;
  align-items: center;
`

export const imageContainer = styled.View`
  flex-direction: row;
  align-items: center;
`

export const ratingText = styled.Text`
  color: ${({ theme }) => theme.colors.black};
  align-self: center;
  font-size: 20px;
  font-weight: 500;
`
