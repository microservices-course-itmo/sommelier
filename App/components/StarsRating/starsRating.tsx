import React from 'react'
import * as Styled from './starsRating.styled'
import filledStar from '../../../assets/images/filledStar.svg'
import unfilledStar from '../../../assets/images/unfilledStar.svg'

type Props = {
  rating: number
  score?: number
}

const preparedStars = (rating: number) => {
  const stars = []
  for (let i = 1; i <= 5; i += 1) {
    if (i > rating) {
      stars.push(<filledStar width='25' height='25' key={i} />)
    } else {
      stars.push(<unfilledStar width='25' height='25' key={i} />)
    }
  }
  return stars
}

const StarsRating = ({ score, rating }: Props) => (
  <Styled.Container>
    <Styled.RatingContainer>
      <Styled.ImageContainer>{preparedStars(rating)}</Styled.ImageContainer>
      {!!score && <Styled.RatingText>{score}</Styled.RatingText>}
    </Styled.RatingContainer>
  </Styled.Container>
)

export default StarsRating
