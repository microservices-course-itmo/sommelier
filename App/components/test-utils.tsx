import React from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const themeRender = (component: any) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
