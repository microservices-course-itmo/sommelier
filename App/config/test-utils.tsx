import React, { ReactNode } from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { theme } from '../theme'

const themeRender = (component: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
