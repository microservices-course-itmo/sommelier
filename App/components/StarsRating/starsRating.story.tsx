import React from 'react'
import { storiesOf } from '@storybook/react-native/dist'
import { withKnobs, number } from '@storybook/addon-knobs'
import { StarsRating } from './starsRating'
import { BufferView, ThemeDecorator } from '../../config/decorators'

const defaultRating = () => (
  <StarsRating score={number('score', 12390)} rating={number('rating', 3.5)} />
)

storiesOf('Stars Rating', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .addDecorator(withKnobs)
  .add('Stars Rating', defaultRating)
