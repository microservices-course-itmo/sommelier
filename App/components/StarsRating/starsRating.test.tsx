import React from 'react'
import { themeRender } from '../test-utils'
import StarsRating from './starsRating'

describe('Stars rating component', () => {
  it('Component renders correctly', () => {
    const { debug } = themeRender(<StarsRating rating={3.5} />)
    debug()
  })
  // it('Rating prop assigns correctly', () => {
  //   const { debug } = themeRender(<StarsRating rating={1} />)
  //   debug()
  // })
})
