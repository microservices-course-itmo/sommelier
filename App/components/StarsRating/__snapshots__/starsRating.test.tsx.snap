// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars rating component Component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    />
  </View>
</View>
`;
